api/android/api/src/main/jni/nnstreamer-native-api.c

```diff
@@ -635,6 +635,106 @@ nns_get_nnfw_type (jint fw_type, ml_nnfw_type_e * nnfw)
   return is_supported && ml_nnfw_is_available (*nnfw, ML_NNFW_HW_ANY);
 }
 
+/**
+ * @brief Set additional environment
+ */
+static gboolean
+nns_set_env (JNIEnv * env, jobject context)
+{
+  gboolean ret = TRUE;
+
+#if defined (ENABLE_SNPE)
+  gboolean snpe_failed = TRUE;
+
+  jclass context_class = NULL;
+  jmethodID get_application_info_method_id = NULL;
+  jobject application_info_object = NULL;
+  jfieldID native_library_dir_field_id = NULL;
+  jstring native_library_dir_path = NULL;
+
+  const gchar *native_library_dir_path_str;
+  gchar *new_path;
+
+  context_class = (*env)->GetObjectClass (env, context);
+  if (!context_class) {
+    nns_loge ("Failed to get context class.");
+    goto snpe_done;
+  }
+
+  get_application_info_method_id = (*env)->GetMethodID (env, context_class,
+      "getApplicationInfo", "()Landroid/content/pm/ApplicationInfo;");
+  if (!get_application_info_method_id) {
+    nns_loge ("Failed to get method ID for `ApplicationInfo()`.");
+    goto snpe_done;
+  }
+
+  application_info_object = (*env)->CallObjectMethod (env, context,
+      get_application_info_method_id);
+  if ((*env)->ExceptionCheck (env)) {
+    (*env)->ExceptionDescribe (env);
+    (*env)->ExceptionClear (env);
+    nns_loge ("Failed to call method `ApplicationInfo()`.");
+    goto snpe_done;
+  }
+
+  native_library_dir_field_id = (*env)->GetFieldID (env,
+      (*env)->GetObjectClass (env, application_info_object),
```

Comment:
you may call DeleteLocalRef() for app-info class

```
app_info_class = (*env)->GetObjectClass (env, application_info_object);

native_library_dir_field_id = (*env)->GetFieldID (env, app_info_class,
......

if (app_info_class)
  (*env)->DeleteLocalRef (env, app_info_class);
```

Reply:
You're right, I missed. I applied your review, please check!